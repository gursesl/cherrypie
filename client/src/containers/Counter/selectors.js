import { createSelector } from 'reselect'
import { SELECTOR_COUNT, SELECTOR_COUNT_VALUE } from './constants'

// Direct selector to state domain
const selectCounter = state => state.get(SELECTOR_COUNT)

// Select the counter value
const makeSelectValue = () =>
  createSelector(selectCounter, counterState => counterState.get(SELECTOR_COUNT_VALUE))

export { selectCounter, makeSelectValue }
