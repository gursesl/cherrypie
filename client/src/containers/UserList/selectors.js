import { createSelector } from 'reselect'
import { SELECTOR_USERS, SELECTOR_USERS_USERS, SELECTOR_USERS_ERROR, SELECTOR_USERS_ISLOADING } from './constants'

// Direct selector to state domain
const selectUsers = state => state.get(SELECTOR_USERS)

// Select users
const makeSelectUsers = () => createSelector(
  selectUsers,
  users => users.get(SELECTOR_USERS_USERS),
)

// Select error
const makeSelectError = () => createSelector(
  selectUsers,
  errorState => errorState.get(SELECTOR_USERS_ERROR),
)

// Select error
const makeSelectIsLoading = () => createSelector(
  selectUsers,
  isLoading => isLoading.get(SELECTOR_USERS_ISLOADING),
)

export {
  selectUsers,
  makeSelectUsers,
  makeSelectError,
  makeSelectIsLoading,
}
