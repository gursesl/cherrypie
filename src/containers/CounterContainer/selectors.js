// import { createSelector } from 'reselect'

// const selectCounterContainerDomain = () => state => state.get('counterContainer')

// const selectCounterContainer = () => createSelector(
//   selectCounterContainerDomain(),
//   (substate) => substate.toJS()
// )

// export default selectCounterContainer

// export {
//   selectCounterContainerDomain
// }
