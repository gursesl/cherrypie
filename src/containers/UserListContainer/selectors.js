import { createSelector } from 'reselect'
import { SELECTOR_USERS, SELECTOR_USERS_USERS, SELECTOR_USERS_ERROR, SELECTOR_USERS_ISLOADING } from './constants'

// Direct selector to state domain
const selectUsers = (state) => state.get(SELECTOR_USERS)

// Select users
const makeSelectUsers = () => createSelector(
  selectUsers,
  (users) => users.get(SELECTOR_USERS_USERS),
  // (errorState) => errorState.get(SELECTOR_USERS_ERROR),
  // (loadingState) => loadingState.get(SELECTOR_USERS_ISLOADING),
)

// Select error
const makeSelectError = () => createSelector(
  selectUsers,
  // (error) => users.get(SELECTOR_USERS_USERS),
  (errorState) => errorState.get(SELECTOR_USERS_ERROR),
  // (loadingState) => loadingState.get(SELECTOR_USERS_ISLOADING),
)

// Select error
const makeSelectIsLoading = () => createSelector(
  selectUsers,
  // (error) => users.get(SELECTOR_USERS_USERS),
  // (errorState) => errorState.get(SELECTOR_USERS_ERROR),
  (isLoading) => isLoading.get(SELECTOR_USERS_ISLOADING),
)

export {
  selectUsers,
  makeSelectUsers,
  makeSelectError,
  makeSelectIsLoading,
}
