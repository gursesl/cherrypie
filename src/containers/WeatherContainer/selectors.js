import { createSelector } from 'reselect'
import {
  SELECTOR_WEATHER,
  SELECTOR_WEATHER_ZIP,
  SELECTOR_WEATHER_CITY,
  SELECTOR_WEATHER_RESULTS,
  SELECTOR_WEATHER_ERROR,
} from './constants'

// Direct selector to Weather domain
const selectWeather = state => state.get(SELECTOR_WEATHER)

const makeSelectZip = () => createSelector(
  selectWeather,
  state => state.get(SELECTOR_WEATHER_ZIP)
)

const makeSelectCity = () => createSelector(
  selectWeather,
  state => state.get(SELECTOR_WEATHER_CITY)
)

const makeSelectResults = () => createSelector(
  selectWeather,
  state => state.get(SELECTOR_WEATHER_RESULTS)
)

const makeSelectError = () => createSelector(
  selectWeather,
  state => state.get(SELECTOR_WEATHER_ERROR)
)

export {
  selectWeather,
  makeSelectZip,
  makeSelectCity,
  makeSelectResults,
  makeSelectError,
}
