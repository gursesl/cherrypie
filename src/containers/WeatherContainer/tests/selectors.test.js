// REFERENCE: SELECTOR TEST
import { fromJS } from 'immutable'
import * as s from '../selectors'
import {
  SELECTOR_WEATHER_ZIP,
  SELECTOR_WEATHER_CITY,
  SELECTOR_WEATHER_RESULTS,
  SELECTOR_WEATHER_ERROR,
} from '../constants'
import * as m from './mockdata'

const rootSelector = s.selectWeather(m.mockedState)

describe('WatherContainer:selectors', () => {
  it('should select the global state', () => {
    expect(rootSelector).toEqual(m.subState)
  })

  it('should select ZIP selector with makeSelectZip()', () => {
    const zipSelector = s.makeSelectZip()(m.mockedState)
    expect(rootSelector.get(SELECTOR_WEATHER_ZIP)).toBe('20165')
    expect(zipSelector).toEqual(m.subState.get(SELECTOR_WEATHER_ZIP))
  })

  it('should select CITY selector with makeSelectCity()', () => {
    const zipSelector = s.makeSelectCity()(m.mockedState)
    expect(rootSelector.get(SELECTOR_WEATHER_CITY)).toBe('Sterling')
    expect(zipSelector).toEqual(m.subState.get(SELECTOR_WEATHER_CITY))
  })

  it('should select RESULTS selector with makeSelectResults()', () => {
    const zipSelector = s.makeSelectResults()(m.mockedState)
    expect(rootSelector.get(SELECTOR_WEATHER_RESULTS)).toBe(fromJS({}))
    expect(zipSelector).toEqual(m.subState.get(SELECTOR_WEATHER_RESULTS))
  })

  it('should select ERROR selector with makeSelectError()', () => {
    const zipSelector = s.makeSelectError()(m.mockedState)
    expect(rootSelector.get(SELECTOR_WEATHER_ERROR)).toBe('Load error')
    expect(zipSelector).toEqual(m.subState.get(SELECTOR_WEATHER_ERROR))
  })

  it('handles the rootSelector snapshot', () => {
    expect(rootSelector).toMatchSnapshot()
  })

  it('handles the SELECTOR_WEATHER_ZIP snapshot', () => {
    expect(rootSelector.get(SELECTOR_WEATHER_ZIP)).toMatchSnapshot()
  })

  it('handles the SELECTOR_WEATHER_CITY snapshot', () => {
    expect(rootSelector.get(SELECTOR_WEATHER_CITY)).toMatchSnapshot()
  })

  it('handles the SELECTOR_WEATHER_RESULTS snapshot', () => {
    expect(rootSelector.get(SELECTOR_WEATHER_RESULTS)).toMatchSnapshot()
  })

  it('handles the SELECTOR_WEATHER_ERROR snapshot', () => {
    expect(rootSelector.get(SELECTOR_WEATHER_ERROR)).toMatchSnapshot()
  })
})
