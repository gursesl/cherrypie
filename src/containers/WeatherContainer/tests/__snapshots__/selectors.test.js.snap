// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WatherContainer:selectors handles the SELECTOR_WEATHER_CITY snapshot 1`] = `"Sterling"`;

exports[`WatherContainer:selectors handles the SELECTOR_WEATHER_ERROR snapshot 1`] = `"Load error"`;

exports[`WatherContainer:selectors handles the SELECTOR_WEATHER_RESULTS snapshot 1`] = `Immutable.Map {}`;

exports[`WatherContainer:selectors handles the SELECTOR_WEATHER_ZIP snapshot 1`] = `"20165"`;

exports[`WatherContainer:selectors handles the rootSelector snapshot 1`] = `
Immutable.Map {
  "app/WeatherContainer/SELECTOR_WEATHER_ZIP": "20165",
  "app/WeatherContainer/SELECTOR_WEATHER_CITY": "Sterling",
  "app/WeatherContainer/SELECTOR_WEATHER_RESULTS": Immutable.Map {},
  "app/WeatherContainer/SELECTOR_WEATHER_ERROR": "Load error",
}
`;
